packages/storage-service/src/InMemoryStorageAdapter.ts

```diff
@@ -0,0 +1,119 @@
+import type { Json } from '@metamask/utils';
+
+import type { StorageAdapter } from './types';
+import { STORAGE_KEY_PREFIX } from './types';
+
+/**
+ * In-memory storage adapter (default fallback).
+ * Implements the {@link StorageAdapter} interface using a Map.
+ *
+ * ⚠️ **Warning**: Data is NOT persisted - lost on restart.
+ *
+ * **Suitable for:**
+ * - Testing (isolated, no mocking needed)
+ * - Development (quick start, zero config)
+ * - Temporary/ephemeral data
+ *
+ * **Not suitable for:**
+ * - Production (unless data is truly ephemeral)
+ * - Data that needs to persist across restarts
+ *
+ * @example
+ * ```typescript
+ * const adapter = new InMemoryStorageAdapter();
+ * await adapter.setItem('SnapController', 'snap-id:sourceCode', 'const x = 1;');
+ * const value = await adapter.getItem('SnapController', 'snap-id:sourceCode'); // 'const x = 1;'
+ * // After restart: data is lost
+ * ```
+ */
+export class InMemoryStorageAdapter implements StorageAdapter {
+  // Explicitly implement StorageAdapter interface
+  /**
+   * Internal storage map.
+   */
+  readonly #storage: Map<string, string>;
+
+  /**
+   * Constructs a new InMemoryStorageAdapter.
+   */
+  constructor() {
+    this.#storage = new Map();
+  }
+
+  /**
+   * Retrieve an item from in-memory storage.
+   * Deserializes JSON data from storage.
+   *
+   * @param namespace - The controller namespace.
+   * @param key - The data key.
+   * @returns The parsed JSON data, or null if not found.
+   */
+  async getItem(namespace: string, key: string): Promise<Json | null> {
+    const fullKey = `${STORAGE_KEY_PREFIX}${namespace}:${key}`;
+    const serialized = this.#storage.get(fullKey);
+
+    if (!serialized) {
+      return null;
+    }
+
+    try {
+      return JSON.parse(serialized) as Json;
```

Comment:
Unnecessary type cast:

```suggestion
      return JSON.parse(serialized);
```